### PR TITLE
WRN-2132: Replaced recompose usages

### PR DESCRIPTION
**Pull request by MikyungKim**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
`recompose` has not been updated over 3 years and its dependencies have been made some troubles.  
So, we'd like to replace `recompose` with other modules.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Replaced recompose usages with internal module and react supported module.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRN-2132

### CODE REVIEW

File: Heading/Heading.js [hunk 1]
```diff
@@ -110,17 +108,27 @@ const HeadingBase = kind({
...
 	Skinnable
 );
 
+HeadingDecorator.PropTypes = /** @lends sandstone/Heading.HeadingDecorator.prototype */ {
+	/**
+	 * Determines what triggers the header content to start its animation. Valid values are
+	 * `'hover'` and `'render'`. The default is `'render'`.
+	 *
+	 * @type {String}
+	 * @default 'render'
```
Comment (jeonghee27) [inline Heading/Heading.js:122]: Is there any reason the type is different with ui/MarqueeDecorator?
```suggestion
	 * Determines what triggers the header content to start its animation.
	 *
	 * @type {('focus'|'hover'|'render')}
	 * @default 'render'
```

Comment (MikyungKim) [reply Heading/Heading.js:122]: No reason :) I just copied it from another place. I'll replace it.

File: internal/util/util.js [hunk 2]
```diff
@@ -51,7 +52,29 @@ const compareChildren = (a, b) => {
...
+ * Updates component only when given props are not shallowly equivalent, not updating otherwise.
+ *
+ * @function
+ * @param   {any}    wrapped    A component
+*  @param   {Array}  propKeys   Prop keys to compare
+ *
+ * @returns {any}               Conditionally memoized component
+ * @memberof sandstone/internal/util
+ * @private
+ */
```
Comment (jeonghee27) [inline internal/util/util.js:65]: Should we give white space between letters like this?

**Review by jeonghee27 — APPROVED** [review]
LGTM

**Pull request merged**